cypress/support/commands.js

```diff
@@ -0,0 +1,149 @@
+        const randomNumber = getRandomInt(0, listing.length - 1)
+        return cy.get(optionSelector).eq(randomNumber).invoke('text').then(text => {
+            return cy.get(menuSelector).select(text).then(() => {
+                return text
+            })
+        })
+    })
+})
+
+Cypress.Commands.add('selectRandomCharacteristics', (fileName) => {
+    const options = {}
+
+    cy.selectRandomOption(Selectors.makeMenuOption, Selectors.makeMenu)
+        .then(text => {
+            options.make = text
+            return text
+        })
+        .then(() => cy.selectRandomOption(Selectors.modelMenuOption, Selectors.modelMenu))
+        .then(text => {
+            options.model = text
+            return text
+        })
+        .then(() => cy.selectRandomOption(Selectors.yearMenuOption, Selectors.yearMenu))
+        .then(text => {
+            options.year = text
+            return text
+        })
+        .then(() => {
+            const filePath = 'cypress/fixtures/' + fileName
+            cy.writeFile(filePath, options)
+        })
+})
+
+Cypress.Commands.add('checkOptionsAreSelected', () => {
+    cy.get(Selectors.noOptionSelectedInMakeMenu).should('not.exist')
+})
+
+Cypress.Commands.add('clickResearchButton', () => {
+    cy.get(Selectors.researchButton).click()
+})
+
+Cypress.Commands.add('checkPageWithCarDescriptionOpens', (filePath) => {
+    cy.fixture(filePath).then(options => {
+        const selector = `${Selectors.selectedOptionsTitle1}:contains(${options.make}), ${Selectors.selectedOptionsTitle2}:contains(${options.make})`
+        cy.get(selector).should('exist').and('be.visible')
+    })
+})
+
+Cypress.Commands.add('selectBasicTrim', () => {
+    cy.get(Selectors.trimOptions).children().eq(0).find('a').click()
+})
+
+Cypress.Commands.add('storeCarCharacteristics', (fileName) => {
+    const carInfo = {}
+    cy.get(Selectors.carCharacteristics).contains(testData.doorCharacteristic).then(($element) => {
+        carInfo.door = $element.text().trim()
+    })
+    cy.get(Selectors.carCharacteristics).contains(testData.seatCharacteristic).then(($element) => {
+        carInfo.seat = $element.text().trim()
+    })
+    cy.get(Selectors.carCharacteristics).contains(/Engine|Motor/i).then(($element) => {
+        carInfo.engine = $element.text().trim()
+    }).then(() => {
+        const filePath = testData.files + fileName
+        cy.writeFile(filePath, carInfo)
+    })
+})
+
+Cypress.Commands.add('goToMainPage', () => {
+    cy.visit(testData.carsMainPage, { headers: { "Accept-Encoding": "gzip, deflate" } })
+})
+
+Cypress.Commands.add('goToCompareTwoCarsLink', () => {
+    cy.get(Selectors.compareTwoCars).click()
+})
+
+Cypress.Commands.add('checkComparePageIsOpen', () => {
+    cy.url().should('contain', testData.comparePage)
+    cy.get(Selectors.comparePageTitle).should('be.visible')
+})
+
+Cypress.Commands.add('addCar1Characteristics', (fileName) => {
+    cy.fixture(fileName).then((carInfo) => {
```

Comment:
It is about handling this value in the file. If you used cy, wrap, you would pass object carInfo as an argument.